classic-commerce.php

```diff
@@ -58,8 +79,64 @@ function cc_wc_already_active_notice() {
 	unset( $_GET['activate'] );
 
 	// Do not proceed further with Classic Commerce loading.
+	$_cc_can_load = false;
+
+} else if (
+	// Check if this is a request that would activate WooCommerce.  Since
+	// Classic Commerce is already active then we also need to prevent
+	// WooCommerce from being activated, again to avoid fatal errors.
+	//
+	// Plugin activation happens after plugins load and after `init` so we need
+	// to check for the presence of the related request parameters here.
+	//
+	// See also src/wp-admin/plugins.php in core.
+	is_admin() &&
+	strpos( $_SERVER['REQUEST_URI'], '/plugins.php' ) !== false &&
+	( isset( $_REQUEST['action'] ) || isset( $_REQUEST['action2'] ) ) &&
+	// Make sure we are really looking at WooCommerce and not the compatibility plugin!
+	file_exists( WP_PLUGIN_DIR . '/woocommerce/includes/class-woocommerce.php' )
+) {
+	$is_activate_woo_request = false;
+
+	// Check if the user tried to activate WooCommerce by itself.
+	if (
+		isset( $_GET['action'] ) &&
+		$_GET['action'] === 'activate' &&
+		isset( $_GET['plugin'] ) &&
+		$_GET['plugin'] === 'woocommerce/woocommerce.php'
+	) {
+		$is_activate_woo_request = true;
+	}
+
+	// Check if the user tried to activate WooCommerce using either of the two
+	// "Bulk Actions" dropdown boxes.
+	if (
+		(
+			( isset( $_POST['action'] ) && $_POST['action'] === 'activate-selected' ) ||
+			( isset( $_POST['action2'] ) && $_POST['action2'] === 'activate-selected' )
+		) &&
+		isset( $_POST['checked'] ) &&
+		in_array( 'woocommerce/woocommerce.php', (array) wp_unslash( $_POST['checked'] ) )
```

Comment:
This part mirrors the logic here: https://github.com/ClassicPress/ClassicPress/blob/1.1.2+dev/src/wp-admin/plugins.php#L73-L79

Reply:
Looks good to me.

![image](https://user-images.githubusercontent.com/4199514/72286981-2a134680-363e-11ea-9ce8-1b400a05fd80.png)

It's not that ugly :)

That's a nice fix!

Reply:
Yes, the end result is fine, but the internals are definitely not ideal.

I looked for a filter that might work to block a plugin activation but didn't find one. Also, activating a plugin happens fairly late in the load order, but we need to get in before the `admin_notices` action at least.

Thanks for taking a look!